uwblueprint/focus-on-nature
# feat: confirmation email for moving between camp sessions

## Notion ticket link

[Confirmation emails for when a camper was moved between campsessions](https://www.notion.so/uwblueprintexecs/Confirmation-emails-for-when-a-camper-was-moved-between-campsessions-e1b8d32a6d8048bfacb8dc2fa16b3511)



## Implementation description
* Added a method in `implementations/emailService.ts` that will send the confirmation email (copy pending approval from FON)
* Called this method in the `updateCampersById` function in `implementations/camperService.ts`



## Steps to test
1. On MongoDB console, find campers and set their primary contact's (first one on the array) email to your email or `focusonnature@uwblueprint.org`. Also make sure that these campers are all registered for the same `campSession` and have the same `chargeId`
2. Send a `PATCH` request to http://localhost:5000/campers/ on Postman. Specify an arbitrary number of `camperId`s to update and make sure to pass `campSession` as a field to be updated. See below for a sample formatting for the body
```
{
    "camperIds": [
        "625786928d4e72b0e41193a6",
        "62576da892c3a430022c653d"
    ],
    "campSession": "628008a06914fb21f29cdd04",
    // add any other fields to be updated

}
```
3. Assuming that the request goes through, check that you receive an email (to the email specified in step 1) that matches the copy in this doc https://docs.google.com/document/d/1LakHl3x4kyR6O6ZhVduJMi6SUqKwyuT99mkFIUbGpQw/edit (email title is: Successfully Moving Camper(s) Between Camp Sessions) and the data in the DB
4. If there is an issue with the `campSessions` or `camp`, an error should be thrown.
5. Test cases: updating 1, 2+ campers' `campSession` (please check that the copy and formatting for the copy is good especially regarding the plurals with multiple campers)
6. Currently there is no logic to handle other errors in the workflow (i.e. if the primary contact's email address doesn't work).



## What should reviewers focus on?
* Code quality - is there a better way to write these methods?
* Any other test cases?


## Checklist
